jobmaster/dm/metadata/state.go

```diff
@@ -0,0 +1,129 @@
+package metadata
+
+import (
+	"context"
+	"encoding/json"
+	"sync"
+
+	"github.com/pingcap/errors"
+	clientv3 "go.etcd.io/etcd/client/v3"
+
+	"github.com/hanfei1991/microcosm/pkg/metadata"
+)
+
+// State represents the state which need to be stored in metadata.
+type State interface{}
+
+// Store manages a type of state.
```

Comment:
I want to add more comments like "Store provides the factory, some utility functions and persistence of State"